File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] nums = new String[numbers.length];
        for (int i = 0; i < nums.length; i++){
             nums[i] = numbers[i] + "";
        }

        Arrays.sort(nums, new Comparator<String>() {
            public int compare(String o1, String o2) {
                return (o2 + o1).compareTo(o1 + o2);
            }
        });

        if(nums[0].charAt(0) == '0') return "0";

        for (int i = 0; i < numbers.length; i++){
            answer += nums[i];
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    num  = list(map(str, numbers)) 
    num = sorted(num, key = cmp_to_key(compare))
    if num[0] == "0" : return "0"
    answer = ''.join(num)
    return answer

def compare(x, y):
    if x + y > y + x:
        return -1
    elif x + y < y + x:
        return 1
    else:
        return 0